### PR TITLE
toplevel_info: Fix racy data creation

### PR DESCRIPTION
**Pull request by Drakulix**
Potential fix for #943

### CODE REVIEW

**Comment by ids1024** [comment]
I no longer see a crash with `wl-paste`, even with `while true; do wl-paste; done`, so it seems like the race condition is gone.

File: src/wayland/protocols/toplevel_info.rs [hunk 1]
```diff
@@ -509,10 +526,14 @@ where
...
-            .iter()
-            .flat_map(|state| (*state as u32).to_ne_bytes())
-            .collect::<Vec<u8>>();
+        let states: Vec<u8> = {
+            let ratio = std::mem::size_of::<States>() / std::mem::size_of::<u8>();
+            let ptr = states.as_mut_ptr() as *mut u8;
+            let len = states.len() * ratio;
+            let cap = states.capacity() * ratio;
+            std::mem::forget(states);
+            unsafe { Vec::from_raw_parts(ptr, len, cap) }
```
Comment (ids1024) [inline src/wayland/protocols/toplevel_info.rs:535]: Why was this changed back to use `Vec::from_raw_parts`?

It is unsound to use `from_raw_parts` with a type that has a different alignment.

Comment (Drakulix) [reply src/wayland/protocols/toplevel_info.rs:535]: rebase mistake. will remove before merge
